src/FakeItEasy/Expressions/ParsedCallExpression.cs

```diff
@@ -34,5 +34,28 @@ internal class ParsedCallExpression
         public IEnumerable<ParsedArgumentExpression> ArgumentsExpressions { get; }
 
         public object CallTarget => this.callTarget.Value;
+
+        // Expression evaluation optimized for a very common case: 
+        // - it's a 'MemberAccess' expression on a local lambda closure (A.CallTo(() => fakeObj.doSomething(args));
+        private static object FastEvaluate(Expression expression)
+        {
+            if (expression == null) return null;
+
+            switch (expression.NodeType)
+            {
+                case ExpressionType.Constant:
+                    return ((ConstantExpression)expression).Value;
+                case ExpressionType.MemberAccess:
+                    var me = ((MemberExpression)expression);
+                    var mi = me.Member;
+                    var fi = mi as FieldInfo;
+                    if (fi != null)
+                    {
+                        return fi.GetValue(FastEvaluate(me.Expression));
+                    }
+                    break;
+            }
+            return expression.Evaluate();
+        }

```

Comment:
While I've got you here. I see you've introduced [a few style warnings](https://ci.appveyor.com/project/FakeItEasy/fakeiteasy/build/87#L503), and on a more "humany" note, we tend to shy away from the short (two-letter) variable names. If you're interested in editing these (especially if you're back here making other changes), I'd consider it a favour.
Otherwise we can fix them in a compensating or superseding commit.